opm/material/fluidsystems/blackoilpvt/LiveOilPvt.hpp

```diff
@@ -579,9 +579,15 @@ class LiveOilPvt
                                                /*temperature=*/Scalar(1e30),
                                                pSat);
 
-            std::pair<Scalar, Scalar> val(Rs, pSat);
+            Pair val(Rs, pSat);
             pSatSamplePoints.push_back(val);
         }
+
+        //Prune duplicate Rs values (can occur, and will cause problems in further interpolation)
+        auto x_coord_comparator = [](const Pair& a, const Pair& b) { return a.first == b.first; };
+        auto last = std::unique(pSatSamplePoints.begin(), pSatSamplePoints.end(), x_coord_comparator);
+        pSatSamplePoints.erase(last, pSatSamplePoints.end());
```

Comment:
I mean the former, but I'm not sure because I consider having to eliminating such duplicates to be quite a hack. (i.e., IMO the input values can be considered to be broken if the x-values are not strictly monotonic in the context of these classes.)